### PR TITLE
futhark: build and install manpages

### PR DESCRIPTION
**Pull request by athas**
###### Motivation for this change

The `futhark` package used to not install man pages.  Now it does.  The only complication is that the `futhark` source tarball contains only manpage *sources* (in RST format), which must be built using Sphinx.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 1]
```diff
@@ -572,17 +572,36 @@ self: super: builtins.intersectAttrs super {
...
+            wrapProgram $out/bin/futhark \
+              --prefix PATH : "${path}" \
+              --set NIX_CC_WRAPPER_x86_64_unknown_linux_gnu_TARGET_HOST 1 \
+              --set NIX_CFLAGS_COMPILE "-I${opencl-headers}/include" \
+              --set NIX_CFLAGS_LINK "-L${ocl-icd}/lib"
+            '';
+    in overrideCabal (addBuildTool super.futhark makeWrapper)
+      (_drv: {
+        buildDepends = [python37Packages.sphinx];
+
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-nix.nix]:
```suggestion
    in overrideCabal (addBuildTools super.futhark [makeWrapper python37Packages.sphinx])
      (_drv: {
```

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 2]
```diff
@@ -572,17 +572,36 @@ self: super: builtins.intersectAttrs super {
...
+        postBuild = ''
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-nix.nix]:
```suggestion
        postBuild = (_drv.postBuild or "") + ''
```

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 3]
```diff
@@ -572,17 +572,36 @@ self: super: builtins.intersectAttrs super {
...
+        make -C docs man
+        '';
+
+        postInstall = ''
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-nix.nix]:
```suggestion
        postInstall = (_drv.postInstall or "") + ''
```

**Review by cdepillabout — CHANGES_REQUESTED** [review]
I left a few comments above with small changes to make, although I haven't actually tested that they work.  If you merge them in, please test that they all actually do what they appear to do.

Also, please rebase this on the `haskell-updates` branch and set the base branch to `haskell-updates` here on github.

**Comment by athas** [comment]
Alright, I made the changes, rebased on haskell-updates, and it still builds fine.  Looks like changing the base branch on GitHub made poor ofborg think this is a much bigger deal than it really is, though.

**Comment by cdepillabout** [comment]
Looks good, and builds on linux, so I'll go ahead and merge this in.

Thanks for fixing this up!

**Pull request merged**